### ISSUE TITLE
Properly typing resolvers

### ISSUE THREAD

**Issue by savovs** [Urigo/graphql-modules#578]
Hey y'all, love this package. I'm trying to figure out how to type my resolvers using codegen but I'm getting this error:

```
Argument of type '{ typeDefs: DocumentNode; resolvers: Resolvers<ModuleContext<{ [key: string]: any; }>>; }' is not assignable to parameter of type 'GraphQLModuleOptions<any, any, any, IResolvers<any, any>>'.
  Types of property 'resolvers' are incompatible.
    Type 'Resolvers<ModuleContext<{ [key: string]: any; }>>' is not assignable to type 'IResolvers<any, any> | IResolvers<any, any>[] | Promise<IResolvers<any, any> | IResolvers<any, any>[]> | ((module: GraphQLModule<any, any, any, IResolvers<any, any>>, ...args: any[]) => MaybePromise<...>) | undefined'.
      Type 'Resolvers<ModuleContext<{ [key: string]: any; }>>' is not assignable to type 'IResolvers<any, any>'.
        Property 'Post' is incompatible with index signature.
          Type 'PostResolvers<ModuleContext<{ [key: string]: any; }>, Post> | undefined' is not assignable to type '(() => any) | GraphQLScalarType | IEnumResolver | IResolverObject<any, any, any> | IResolverOptions<any, any, any>'.
            Type 'undefined' is not assignable to type '(() => any) | GraphQLScalarType | IEnumResolver | IResolverObject<any, any, any> | IResolverOptions<any, any, any>'.
```

Here's what I'm doing:
**codegen.yml**
```yml
overwrite: true
schema: src/**/*.schema.ts
require:
  - ts-node/register/transpile-only # required if you're using TS-Node
generates:
  ./src/modules/types.ts:
    plugins:
      - typescript
      - typescript-resolvers
    config:
      contextType: '@graphql-modules/core#ModuleContext'
```

**user.resolvers.ts**
```ts
import { Resolvers } from '../types'
import { getMockPost, getMockUser } from '../../mocks'

const resolvers: Resolvers = {
	Post: {
		id: post => post.id,
		title: post => post.title || '',
		body: post => post.body || '',
		user: getMockUser
	},
	User: {
		posts: () => [getMockPost()]
	},
	Query: {
		allPosts: () => [getMockPost()]
	}
}

export default resolvers
```

**user.module.ts**
```ts
import { GraphQLModule } from '@graphql-modules/core'

import typeDefs from './post.schema'
import resolvers from './post.resolvers'

export default new GraphQLModule({
	typeDefs,
	resolvers
})
```

It works if I do `resolvers: any` but that kinda defeats the purpose of having the types in the first place.

Here's a reproduction branch of the repo: https://github.com/savovs/ts-gql/tree/type-resolvers

Anyone got an idea how to fix this?

**Comment by kamilkisiela** [comment]
@savovs 

#597

https://github.com/savovs/ts-gql/pull/1

**Comment by kamilkisiela** [comment]
Released the fix in `v0.7.9`
